### ISSUE TITLE
Format validation missing from `getCurrentTexture`

### ISSUE THREAD

**Issue by kainino0x** [gpuweb/gpuweb#4906]
Reported by @sagudev:

> There is no validation for https://www.w3.org/TR/webgpu/#supported-context-formats in `getCurrentTexture`, so it will always return valid texture (if desc valid otherwise) and not raise validation error.

I think we just need to add this validation to getCurrentTexture and make it return an invalid texture (like it would if the configuration was invalid for any other reason.

**Comment by sagudev** [comment]
This means that we need to tell `GPUTexture` that it was created from getCurrentTexture.

**Comment by kainino0x** [comment]
@sagudev Where would that be needed? I was thinking just something like this: https://github.com/gpuweb/gpuweb/pull/4907

**Comment by greggman** [comment]
I thought you already get an error from `configure`

https://www.w3.org/TR/webgpu/#dom-gpucanvascontext-configure

Is this handling a case not covered there?

**Comment by greggman** [comment]
Conversations here shouldn't be referring to ephemeral chats. Can you summarize the issue?

> The problem is that we only raise validation error if format is not in https://www.w3.org/TR/webgpu/#supported-context-formats then and canvas context is still configured.

You lost me

As I see it

1. You don't call `configure`  - `getCurrentTexture` fails
2. You call configure with a valid format - `getCurrentTexture` succeeds
3. you call configure with an invalid format - `configure` fails. `getCurrentTexture` does (1) or (2) since a failed call to `configure` is a no-op meaning the canvas is still either (a) unconfigured or (b) still configured to it's previous valid state

How is it possible to call `getCurrentTexture` with the wrong format? It gets it's format from `configure` and that format is already validated.

**Comment by sagudev** [comment]
The problem is that if you configure with invalid format (not one of supported context formats), canvas still gets configured and validation error is raised. Because context is configured `getCurrentTexture` returns valid texture, but given that it hasn't got supported context formats I am wondering what to do with it.

**Comment by greggman** [comment]
> The problem is that if you configure with invalid format (valid per device but not one of supported context formats), canvas still gets configured and validation error is raised. Because context is configured `getCurrentTexture` returns valid texture, but given that it hasn't got supported context formats I am wondering what to do with it.

That seems like a bug in the spec.  The canvas should be configured with an invalid format. The bug is is that this happens on the device timeline, not the content timeline. Fix that, and then `getCurrentTexture` doesn't need to be changed. Is it impossible to move that validate to `configure` on the content timeline?

**Comment by sagudev** [comment]
> > The problem is that if you configure with invalid format (valid per device but not one of supported context formats), canvas still gets configured and validation error is raised. Because context is configured `getCurrentTexture` returns valid texture, but given that it hasn't got supported context formats I am wondering what to do with it.
> 
> That seems like a bug in the spec. The canvas not should be configured with an invalid format. The bug is is that this happens on the device timeline, not the content timeline. Fix that, and then `getCurrentTexture` doesn't need to be changed. Is it impossible to move that validate to `configure` on the content timeline?

That's exactly what I proposed in https://github.com/gpuweb/gpuweb/issues/4906#issuecomment-2387648012. I am already writing PR for it.
